shared/cryptography/codeql/cryptography/Model.qll

```diff
@@ -957,9 +983,27 @@ module CryptographyBase<LocationSig Location, InputSig<Location> Input> {
     abstract string getRawHashAlgorithmName();
 
     /**
-     * Gets the length of the hash digest in bits.
+     * Gets the length of the hash digest in bits if it is not an implicit size
+     * and is not fixed by the algorithm.
+     * For example, SHA-256 has a fixed length of 256 bits.
+     * SHA-1 should not be modled with digest length as it is always 160 bits.
+     * Fixed length digests are modeled with `fixedImplicitDigestLength` and
+     * are used at the node level.
      */
-    abstract int getDigestLength();
+    abstract int getFixedDigestLength();
+  }
+
+  predicate fixedImplicitDigestLength(THashType type, int digestLength) {
```

Comment:
This should be handled not in the instance but rather implicitly in the node. Documentation should exist to clarify which algorithms need `getDigestLength()` specified or a global predicate in `Model.qll` should exist as a helper to define those implicit type/digestlen mappings. I suggest a predicate on the node that takes the type from `this`.